### PR TITLE
Feature/96 navigate auth

### PR DESCRIPTION
**Pull request by dakna**
implements different auth states based on the completion state of the user and it's details.  see also https://github.com/NexdApp/nexd-android/issues/96

### CODE REVIEW

**Comment by andre-stefanov** [comment]
Registering a new account leads to the home screen again instead of progressing to the registration details screen.

File: app/src/main/java/app/nexd/android/ui/MainActivity.kt [hunk 1]
```diff
@@ -20,15 +19,15 @@ class MainActivity : AppCompatActivity() {
...
+/*        findNavController(R.id.nav_host_fragment).addOnDestinationChangedListener { controller, destination, _ ->
             Log.v("Navigation", destination.toString())
             runOnUiThread {
                 // skip authentication if user is logged in
                 mainViewModel.getNavigationOverride(destination)?.let {
                         controller.navigate(it)
                 }
             }
-        }
+        }*/
```
Comment (andre-stefanov) [inline app/src/main/java/app/nexd/android/ui/MainActivity.kt:30]: please just remove unused code if it is not needed anymore.

File: app/src/main/java/app/nexd/android/ui/MainViewModel.kt [hunk 2]
```diff
@@ -26,4 +43,60 @@ class MainViewModel(private val preferences: Preferences) : ViewModel() {
         }
```
Comment (andre-stefanov) [inline app/src/main/java/app/nexd/android/ui/MainViewModel.kt:43]: please remove unused functions

**Comment by dakna** [comment]
> Registering a new account leads to the home screen again instead of progressing to the registration details screen.

the idea was to let the first home screen decide if it is needed, it would navigate to the details again, possibly with a message why all those details are still needed. I changed that now, and the user goes directly to the details screen again, only on next login the user would be navigated to details if he skipped that step before by exiting the app.

**Pull request merged**